### PR TITLE
add article about data security of Salesforce

### PR DESCRIPTION
**Pull request by Hazyzh**

### CODE REVIEW

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+In Salesforce you can control which users have access to which data in your **whole org**, a specific **object**, a specific **field**, or an individual **record**.
+
+#### Control Access to the Organization
+
+At the organization level's data security there has one thing you can do is that ensure only employees who meet certain criteria can log in to Salesforce. In Salesforce you can do this by managing authorized users, setting password policies, and limiting when and where users can log in.
+
+#### Control Access to Objects
+
+In Salesforce an object is a collection of records, like leads or contacts. So how to control access to objects? Here we must take about **profile**.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: take about => talk about
profile => profiles (generally you use plural form here)

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+A profile is a collection of settings and permissions. Every user in Salesforce has his own's profile, You can set object permissions with profiles or permission sets to control whether a group of users can create, view, edit, or delete any records of that object.
+
+#### Control Access to Fields
+
+After controlling `object-level` access, defining `field-level` security for sensitive fields is also an important part of Salesforce data security.
+
+In some cases, you want users to have access to an object, but limit their access to individual fields in that object. You can apply field settings by modifying profiles or permission sets just like the control in objects.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: just like ``the access control of``` objects.

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+#### Control Access to Records
+
+As just said, you can let particular users view specific fields in a specific object by set profiles and permission sets. But how to restrict the individual records they're allowed to see?
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: As "I" just said (you cannot leave out the subject in English. Also I'd say just drop this phrase, or use "As I mentioned before" instead.)
by set profiles => by ```setting``` profiles
But how to restrict ```individual records that``` they are allowed to see?

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+In Salesforce you control record-level access in four ways, `Org-wide defaults`, `Role hierarchies`, `Sharing rules` and `Manual sharing`.
+
+![](/integration-blog/assets/2019-02-26-data-security-of-salesforce/data_security_records.jpeg)
+
+`org-wide` specify the default level of access each records, if you defined this setting with *public*, All users can view on the record, but if you set with *private*, users except for the record owner whether can see it was controlled by `Role hierarchies` and `Sharing rules`.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: ... level of access ```for``` each ```record```, if you ```set this to``` *public*, ```all``` users can view ```these records```. ```However,``` if you set ```this to``` *private*, ```whether the users other than the record owner can see the records is``` controlled by `role hierarchies` and `sharing rules`.

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+After controlling `object-level` access, defining `field-level` security for sensitive fields is also an important part of Salesforce data security.
+
+In some cases, you want users to have access to an object, but limit their access to individual fields in that object. You can apply field settings by modifying profiles or permission sets just like the control in objects.
+
+#### Control Access to Records
+
+As just said, you can let particular users view specific fields in a specific object by set profiles and permission sets. But how to restrict the individual records they're allowed to see?
+
+In Salesforce you control record-level access in four ways, `Org-wide defaults`, `Role hierarchies`, `Sharing rules` and `Manual sharing`.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: I don't think it is necessary to capitalize org-wide defaults, role hierarchies, sharing rules, and manual sharing.

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+![](/integration-blog/assets/2019-02-26-data-security-of-salesforce/data_security_records.jpeg)
+
+`org-wide` specify the default level of access each records, if you defined this setting with *public*, All users can view on the record, but if you set with *private*, users except for the record owner whether can see it was controlled by `Role hierarchies` and `Sharing rules`.
+
+`Role hierarchies` ensure managers have access to the same records as their subordinates. Each role in the hierarchy represents a level of data access that a user or group of users needs.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: ... a user or ```a``` group of users needs. (Need or needs... either sound rather awkward here.)
Alternatively:
Each role in the hierarchy represents ```the``` level of data access ```for``` a user or a group of users.

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+`Sharing rules` and `Manual sharing` give some user access to records they don’t own or can’t normally see.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: Manual sharing => manual sharing

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+### Example
+
+If company X has 5 sales teams, each team has 1 manager, and also 1 super manager for all managers.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: and also 1 super manager => and there is 1 super manager

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+```
+ Super manager 
+       Manager1 <- Team1
+       Manager2 <- Team2
+       Manager3 <- Team3
+       Manager4 <- Team4
+       Manager5 <- Team5
+```
+
+Profile should be only 3:
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: ```There should be only 3 profiles```:
Alternatively
```There should only be 3 profiles```:
They really mean the same, but the tone is different, pick which ever sound right to you.

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+  - **Sales Team** (Assigned to all Teams)
+  - **Manager** (Assigned to all Managers)
+  - **Super Manager** (Assigned to super manager) 
+
+  
+### Conclusion
+
+Salesforce platform provides a flexible, layered sharing model, it’s easy to assign different data sets to different sets of users but we should think more before we design and implement our data model.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: ... layered sharing model```.``` ```It is``` easy ... ... to different sets of users```.``` ```But``` we should ... ... (Don't use run-on sentences like this. Keep each sentence short and to the point will make your writing sounds stronger.)

Comment (Hazyzh) [reply _posts/2019-02-26-data-security-of-salesforce.md]: ok thanks

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+title:  "Data Security Of Salesforce"
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: Of => of (There are some words that don't have to be capitalized in titles)
Also, I think "Data Security ```in``` Salesforce" sounds a bit better.

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+date:   2019-02-26 22:27:00 +0800
+categories: blog
+author: Harry Hou
+---
+
+[Articles](https://trailhead.salesforce.com/content/learn/modules/data_security)
+
+## Data Security of Salesforce
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: of => in

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+As a CRM application, it is important for the user to give the right people access to the right data. This post will introduce how to manage access control in Salesforce.
+
+### Levels of data access
+
+In Salesforce, you can control the level of access of your users to the data in your whole org, down to specific objects, fields, or even inidividual records.
+
+#### Control Access to the Organization
+
+For data security at organization level, what you can do is that ensure that only employees who meet certain criteria are allowed to log in to Salesforce. You can do this by managing authorized users, setting password policies, and limiting when and where users can log in.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: what you can do is that ensure => what you can do is ```to``` ensure

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+#### Control Access to Objects
+
+In Salesforce an object is a collection of records, like leads or contacts. So how to control access to objects? Here we must talk about **profiles**.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: In Salesforce```,``` an object...

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+A profile is a collection of settings and permissions. Every user in Salesforce has his own's profile, You can set object permissions with profiles or permission sets to control whether a group of users can create, view, edit, or delete any records of that object.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: his own's profile => his ```own``` profile

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+#### Control Access to Fields
+
+After controlling `object-level` access, defining `field-level` security for sensitive fields is also an important part of Salesforce data security.
+
+In some cases, you want users to have access to an object, but limit their access to individual fields in that object. You can apply field settings by modifying profiles or permission sets just like the access control of objects.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: but limit their access => but ```to``` limit their access (this is to keep the tense the same: you want users "to have" access, but "to limit" their access to...)

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+#### Control Access to Records
+
+As I mentioned before, you can let particular users view specific fields in a specific object by setting profiles and permission sets. But how to restrict individual records that they're allowed to see?
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: But how to restrict individual... => But how to ```control the``` individual (I think it makes more sense to use "control" here)

File: _posts/2019-02-26-data-security-of-salesforce.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+
+In Salesforce you control record-level access in four ways, `org-wide defaults`, `role hierarchies`, `sharing rules` and `manual sharing`.
+
+![](/integration-blog/assets/2019-02-26-data-security-of-salesforce/data_security_records.jpeg)
+
+`org-wide` specify the default level of access for each records, if you set this to *public*, all users can view these records. However, if you set this to *private*, whether the users other than the record owner can see the records is controlled by `role hierarchies` and `sharing rules`.
```
Comment (u9520107) [inline _posts/2019-02-26-data-security-of-salesforce.md]: `org-wide` => `Org-wide defaults` (Capitalize first word of sentence, and also make sure that the item is the same as the list above)

**Pull request merged**